daemon/daemon.cc

```diff
@@ -228,14 +228,43 @@ static auto onFileAdded(tr_watchdir_t dir, char const* name, void* vsession)
 {
     auto const* session = static_cast<tr_session const*>(vsession);
 
-    if (!tr_str_has_suffix(name, ".torrent"))
+    if (!tr_str_has_suffix(name, ".torrent") && !tr_str_has_suffix(name, ".magnet"))
     {
         return TR_WATCHDIR_IGNORE;
     }
 
     auto filename = tr_strvPath(tr_watchdir_get_path(dir), name);
     tr_ctor* ctor = tr_ctorNew(session);
-    if (!tr_ctorSetMetainfoFromFile(ctor, filename.c_str(), nullptr))
+
+    bool retry = false;
+
+    if (tr_str_has_suffix(name, ".torrent"))
+    {
+        if (!tr_ctorSetMetainfoFromFile(ctor, filename.c_str(), nullptr))
+        {
+            retry = true;
+        }
+    }
+    else if (tr_str_has_suffix(name, ".magnet"))
```

Comment:
This is always true due to the test up on line 231

```suggestion
    else // ".magnet" suffix
```